wsmaster/che-core-api-devfile/src/main/java/org/eclipse/che/api/devfile/server/convert/tool/PluginToolConverter.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.api.devfile.server.convert.tool;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static java.util.stream.Collectors.toMap;
+import static org.eclipse.che.api.core.model.workspace.config.Command.PLUGIN_ATTRIBUTE;
+import static org.eclipse.che.api.devfile.server.Constants.PLUGINS_TOOLS_ALIASES_WORKSPACE_ATTRIBUTE_NAME;
+import static org.eclipse.che.api.devfile.server.Constants.PLUGIN_TOOL_TYPE;
+import static org.eclipse.che.api.workspace.shared.Constants.WORKSPACE_TOOLING_PLUGINS_ATTRIBUTE;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import org.eclipse.che.api.devfile.model.Action;
+import org.eclipse.che.api.devfile.model.Command;
+import org.eclipse.che.api.devfile.model.Devfile;
+import org.eclipse.che.api.devfile.model.Tool;
+import org.eclipse.che.api.devfile.server.FileContentProvider;
+import org.eclipse.che.api.workspace.server.model.impl.WorkspaceConfigImpl;
+import org.eclipse.che.commons.annotation.Nullable;
+
+/**
+ * Helps to convert chePlugin tools to workspace environments and vice versa.
+ *
+ * @author Sergii Leshchenko
+ */
+public class PluginToolConverter implements ToolConverter {
```

Comment:
Looks like all the new `*converter` classes are not converters but rather appliers/provisioners. Would be cool to refactor them to simplify and make less ambiguous.

Reply:
I believe it is not actual anymore after the latest changes